### PR TITLE
Provide POVSupplier for ManualShooter command

### LINKED ISSUE #75: Provide POVSupplier for ManualShooter command
Rather than passing a joystick in the ctor, pass a POVSupplier.

### PR DESCRIPTION
**Pull request by MatthewZ0823**
Fix #75 

Rather than passing a joystick in the ctor, pass a POVSupplier.

### CODE REVIEW

File: src/main/java/frc/robot/commands/shooter/ManualShooter.java [hunk 1]
```diff
@@ -38,9 +39,10 @@ public void initialize() {
   @Override
   public void execute() {
     m_shooter.set(Constants.Shooter.IDLE_SPEED * Constants.Shooter.VELOCITY_FF); // Run at resting speed
-    if (m_joystick.getPOV() == 0 || m_joystick.getPOV() == 45 || m_joystick.getPOV() == 315) {
+    if (m_POVSupplier.getAsDouble() == 0 || m_POVSupplier.getAsDouble() == 45 || m_POVSupplier.getAsDouble() == 315) {
       m_shooter.setSetpoint(m_shooter.getSetpoint() + 50);
-    } else if (m_joystick.getPOV() == 180 || m_joystick.getPOV() == 135 || m_joystick.getPOV() == 225) {
+    } else if (m_POVSupplier.getAsDouble() == 180 || m_POVSupplier.getAsDouble() == 135
+        || m_POVSupplier.getAsDouble() == 225) {
```
Comment (Jacob4649) [inline src/main/java/frc/robot/commands/shooter/ManualShooter.java]: Not quite what we're looking for, the intent was more to have `m_shooter.setSetpoint(m_setpointSupplier.getAsDouble())`, and then pass the POV logic as an argument

File: src/main/java/frc/robot/commands/shooter/ManualShooter.java [hunk 2]
```diff
@@ -7,26 +7,27 @@
...
-   * 
+   *
    * @param joystick the joystick to use to set the setpoint
    * @param shooter  the shooter to use for this command
    */
-  public ManualShooter(Joystick joystick, Shooter shooter) {
+  public ManualShooter(DoubleSupplier POVSupplier, Shooter shooter) {
     m_shooter = shooter;
-    m_joystick = joystick;
+    m_POVSupplier = POVSupplier;
```
Comment (Jacob4649) [inline src/main/java/frc/robot/commands/shooter/ManualShooter.java]: Rename to `m_speedSupplier`

**Review by Jacob4649 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/shooter/ManualShooter.java (hunk 1 above)
Comment (MatthewZ0823) [reply src/main/java/frc/robot/commands/shooter/ManualShooter.java]: I'm not sure I understand. Which argument do I pass the POV logic to.

File: src/main/java/frc/robot/commands/shooter/ManualShooter.java [hunk 3]
```diff
@@ -7,26 +7,27 @@
...
-  private final Joystick m_joystick;
+  private final DoubleSupplier m_speedSupplier;
 
   /**
    * Sets the setpoint shooter speed manually, and runs the shooter at its resting
    * speed
-   * 
+   *
    * @param joystick the joystick to use to set the setpoint
    * @param shooter  the shooter to use for this command
```
Comment (Jacob4649) [inline src/main/java/frc/robot/commands/shooter/ManualShooter.java]: Update javaDocs

**Review by Jacob4649 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/shooter/ManualShooter.java (hunk 2 above)
Comment (MatthewZ0823) [reply src/main/java/frc/robot/commands/shooter/ManualShooter.java]: Resolved in commit 7dc8382

File: src/main/java/frc/robot/commands/shooter/ManualShooter.java (hunk 1 above)
Comment (MatthewZ0823) [reply src/main/java/frc/robot/commands/shooter/ManualShooter.java]: Resolved in commit 7dc8382

**Review by Jacob4649 — APPROVED** [review]

**Comment by matthew-reynolds** [comment]
This is not what I had in mind, but it works too.

I had imagined creating a `POVSupplier` class, similar to the `DoubleSupplier` class, and leaving the angle logic in the command. Our `RobotContainer` is very bloated and difficult to read with all our big lambdas. We should try to move logic into the commands when it makes sense.

**Pull request merged**